kasa/smart/smartdevice.py

```diff
@@ -305,8 +305,22 @@ async def _initialize_features(self):
             for feat in module._module_features.values():
                 self._add_feature(feat)
 
-    def get_module(self, module_name) -> SmartModule | None:
+    @overload
+    def get_module(self, module_type: type[ModuleT]) -> ModuleT | None: ...
+
+    @overload
+    def get_module(self, module_type: str) -> SmartModule | None: ...
+
+    def get_module(
+        self, module_type: type[ModuleT] | str
+    ) -> ModuleT | SmartModule | None:
         """Return the module from the device modules or None if not present."""
+        if isinstance(module_type, str):
+            module_name = module_type
+        elif issubclass(module_type, SmartModule):
+            module_name = module_type.__name__
+        else:
+            return None
```

Comment:
Open question, is it better to return `None` instead of raising an exception here?

Reply:
As per discussion this lets us do 
```py
if (brightness := dev.get_module("brightness")) is not None:
    await brightness.set_brightness(123)
 ```